android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -25,10 +26,38 @@ class JsonConverter {
          */
         @JvmStatic
         fun toPlayerConfig(json: ReadableMap?): PlayerConfig {
+            var config = PlayerConfig()
             if (json != null && json.hasKey("licenseKey")) {
-                return PlayerConfig(key = json.getString("licenseKey"))
+                config = PlayerConfig(key = json.getString("licenseKey"))
             }
```

Comment:
I would suggest something like this:

```kotlin
if (json == null) return PlayerConfig()
            
val playerConfig = if (json.hasKey("licenseKey")) {
    PlayerConfig(key = json.getString("licenseKey"))
} else {
    PlayerConfig()
}
````

The suggestion also includes a more descriptive name for the "config" variable i.e. `playerConfig`

Reply:
Hi @matamegger @pogist 
At this point, I understand the guidelines but it loses a bit of sense as it is making the instance of PlayerConfig but does not match it to any variable. 

```java
     @JvmStatic
     fun toPlayerConfig(json: ReadableMap?): PlayerConfig {
        var config = PlayerConfig()
        if (json != null && json.hasKey("licenseKey")) {
            config = PlayerConfig(key = json.getString("licenseKey"))
        }
        if (json != null && json.hasKey("playbackConfig")) {
            var playbackConfigJson = json.getMap("playbackConfig")
            if (playbackConfigJson?.hasKey("isAutoplayEnabled") == true) {
                config.playbackConfig.isAutoplayEnabled = playbackConfigJson.getBoolean("isAutoplayEnabled")
            }
            /**
             * Specifies if the player should start muted.
             *
             * Default is `false`.
             */
            if(playbackConfigJson != null && playbackConfigJson.hasKey("isMuted")) {
                config.playbackConfig.isMuted = playbackConfigJson.getBoolean("isMuted")
            }
            /**
             * Specifies if time shifting (during live streaming) should be enabled.
             *
             * Default is `true`.
             */
            if(playbackConfigJson != null && playbackConfigJson.hasKey("isTimeShiftEnabled")) {
                config.playbackConfig.isTimeShiftEnabled = playbackConfigJson.getBoolean("isTimeShiftEnabled")
            }
        }
        return config
    }
```

The original idea is to be able to modify the values of ```config.playbackConfig.isAutoplayEnabled``` with the value received via parameter from RN, in the case that the config variable does not exist, it would not take this value as it is not specified, is this correct? or is this something I am missing?

Reply:
HI @jonathanm-tkf,
I don't quite understand your question.
I will try to elaborate on my suggested code snippet.

The function consists of many `if` statements at the moment. In each of the `if`s we check if `json` is `null`. In the case that `json` is indeed `null`, we basically do nothing and return `config`... which was initialized with a default `PlayerConfig`.
To avoid the many `null` checks I suggested the first line of my snippet:
```kotlin
if (json == null) return PlayerConfig()
```

Furthermore, `PlayerConfig.key` is a `val` and thus can not be changed after initializing the `PlayerConfig`. This means that we directly need to pass the `licenseKey` from the `ReadableMap` to the `PlayerConfig` constructor (if there is a `licenseKey` available in the `ReadableMap`).
The code in the PR already does this.

With my suggestion:
```kotlin
val playerConfig = if (json.hasKey("licenseKey")) {
    PlayerConfig(key = json.getString("licenseKey"))
} else {
    PlayerConfig()
}
```
We **always** assign a value for `playerConfig`. Depending on if there is a `licenseKey` in the `ReadableMap`, we either create a default `PlayerConfig` or a `PlayerConfig` with a player license key.

As mentioned in the first comment, I would like you to rename the current variable name `config` to `playerConfig`. i.e. there should be no local variable called `config` anymore :)

So if you replace:
```kotlin
var config = PlayerConfig()
if (json != null && json.hasKey("licenseKey")) {
    config = PlayerConfig(key = json.getString("licenseKey"))
}
```
with my suggestion, and of course every usage of `config` with `playerConfig`, everything should still work as intended by you. 🙂

Reply:
Hi @matamegger thank you for your support, now I've caught it

apologies for the misunderstanding, I have updated the PR with the resolution as above and have taken the liberty of also removing some nullity checks as you had mentioned above, let me know if any other corrections are needed.

Regards.